src/crimson/CMakeLists.txt

```diff
@@ -129,7 +128,12 @@ add_library(crimson-common STATIC
   $<TARGET_OBJECTS:common_mountcephfs_objs>
   $<TARGET_OBJECTS:common-options-objs>)
 
-target_compile_definitions(crimson-common PRIVATE
+if(WITH_SYSTEMD)
+  target_sources(crimson-common PRIVATE
+    ${PROJECT_SOURCE_DIR}/src/common/Journald.cc)
+endif()
+ 
+  target_compile_definitions(crimson-common PRIVATE
```

Comment:
Thank for spotting this one!
See how `crimson-common` uses the following conditional flags:
```
    $<$<NOT:$<BOOL:${WITH_SYSTEM_BOOST}>>:${ZLIB_LIBRARIES}>
    $<$<BOOL:${WITH_BREAKPAD}>:Breakpad::client>
```
Same could be done with something like:
```
add_library(crimson-common STATIC
    ...files...
    $<$<BOOL:${WITH_SYSTEMD}>:${PROJECT_SOURCE_DIR}/src/common/Journald.cc>
```

Reply:
Thanks, will change to that